### PR TITLE
Added numpy shape function and tests

### PR DESCRIPTION
**Pull request by alvin-98**
- [] #3284

### CODE REVIEW

**Review by RashulChutani — CHANGES_REQUESTED** [review]
Hi @alvin-98 As mentioned in the Contributing guide (https://lets-unify.ai/ivy/contributing/4_open_tasks.html#frontend-apis), we do not need to have type hints for frontend functions. Can we please remove the same? Thanks.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_basic_operations.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+def test_numpy_shape(
+    xs_n_input_dtypes_n_unique_idx,
+    as_variable,
+    num_positional_args,
+    native_array,
+    fw,
+):
+    input_dtypes, xs = xs_n_input_dtypes_n_unique_idx
+    xs = np.asarray(xs, dtype=input_dtypes)
+    print(xs)
```
Comment (roboFiddle) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_basic_operations.py:30]: Please don't print in tests as it clutters the output.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_basic_operations.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        input_dtypes=input_dtypes,
+        as_variable_flags=as_variable,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        with_out=False,
+        fw=fw,
+        frontend="numpy",
+        fn_tree="shape",
+        array=xs,
+    )
```
Comment (roboFiddle) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_basic_operations.py:41]: There should be a empty line at the end of the file.

**Review by roboFiddle — CHANGES_REQUESTED** [review]
I put a couple changes that need to be made, otherwise it looks good.

**Comment by roboFiddle** [comment]
Hi, it seems like there was an issue merging this PR. Could you reopen this or open a new PR and I'll merge it right away. 

Sorry for the difficulty.

**Comment by alvin-98** [comment]
Hi, I have created a new PR here #4029. Please let me know if this works.

**Pull request closed without merging**